src/Core/src/WeakEventManager.cs

```diff
@@ -125,15 +125,23 @@ void RemoveEventHandler(string eventName, object? handlerTarget, MemberInfo meth
 			if (!_eventHandlers.TryGetValue(eventName, out List<Subscription>? subscriptions))
 				return;
 
-			for (int n = subscriptions.Count; n > 0; n--)
+			for (int n = subscriptions.Count - 1; n >= 0; n--)
 			{
-				Subscription current = subscriptions[n - 1];
+				Subscription current = subscriptions[n];
 
-				if (current.Subscriber?.Target != handlerTarget || current.Handler.Name != methodInfo.Name)
+				if (current.Subscriber != null && !current.Subscriber.IsAlive)
```

Comment:
```suggestion
				if (current.Subscriber is not null && !current.Subscriber.IsAlive)
```
@PureWeen has an extension method for isAlive that already check it is not null, could it be used here?

Reply:
Is it this one?

https://github.com/dotnet/maui/blob/236a60aad7188652ad43f9523979b26f5a2790fc/src/Controls/src/Core/WeakReferenceExtensions.cs#L8

In this case, we don't really need the generic version, as it is an `object` that looks like it can also be null.

I think `static` events can have a null `handlerTarget` here:

```csharp
if (current.Subscriber?.Target == handlerTarget && current.Handler.Name == methodInfo.Name)
```